spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java

```diff
@@ -126,6 +134,7 @@ public StreamDeploymentController(StreamDefinitionRepository repository, Deploym
 		this.registry = registry;
 		this.deployer = deployer;
 		this.metadataResolver = metadataResolver;
+		this.appsProperties = commonProperties;
```

Comment:
we should assert this is not null, since otherwise we could hit an NPE on line 309